### PR TITLE
docs(adr,schema): methodology + ADR log + add-a-provider/suite/metric guide (ENG-69)

### PR DESCRIPTION
**Pull request by dbworku**
**Sandbox Benchmark Matrix & Leaderboard — one exploration branch split into a parallel-mergeable dependency DAG.**
Every PR is green on its own (typecheck + tests) and merges bottom-up. Four independent units:

- **Standalone (off `main`):** #76 ENG-59 (lifecycle) · #77 ENG-69 (docs/ADRs)
- **Shared CLI/harness + dimensions foundation:** #67 ENG-62 → #68 ENG-60
- **CI / matrix branch:** #67 → #68 → #69 ENG-66 → #70 ci-hardening
- **Dataset / consumer branch:** #67 → #68 → #71 ENG-61 → #72 ENG-70 → #73 ENG-67 → { #74 ENG-68 ‖ #75 ENG-71 }

↑ **This PR is #77 — ENG-69**, based on `main`.

---

Document how a Run is produced and how to extend the matrix so the comparison is reproducible and extensible by outsiders. Adds docs/methodology.md, CONTRIBUTING.md (local gate commands + add-a-provider/suite/metric walkthroughs), and a 5-record ADR log with index. Schema-side (additive on main): catalog asserts snake_case ids and that no derived metric carries a pts source; analysis exports percentileOf() sharing the R-7 helper the aggregator uses. Independent stack — off main.

Also fixes two authoring defects: strips stray closing tags accidentally committed into all five ADRs plus the ADR index, and repoints a dangling methodology link to ADR-0003.

---
## Summary by cubic
Adds methodology and ADR docs, plus a contributor guide for adding providers/suites/metrics, and tightens the schema with a public `percentileOf()` helper and new catalog invariants. Addresses ENG-69 by documenting how a Run is produced, how to extend the matrix safely, and updating ADR-0005 to disclose `hostCpuModels`.

- **New Features**
  - Docs: `docs/methodology.md`, ADR log (0001–0005 + index), and `CONTRIBUTING.md` with local gate and add-a-provider/suite/metric guides; README links added.
  - Schema (`@sandbox-benchmarks/schema`): exports `percentileOf()` using the same R-7 interpolation as `aggregate`; validates `p` in [0,1] and finite samples; tests cover parity and edge cases.
  - Catalog: enforces snake_case metric ids and forbids `pts` on `derived` metrics; tests added.

- **Bug Fixes**
  - Removed stray closing tags from all ADRs and the ADR index.
  - Fixed a broken methodology link to ADR-0003.

<sup>Written for commit 47043650ccd7327f60c22017c3883539fdfd41a9. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/starslingdev/sandbox-benchmarks/pull/77?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

---

**Verification:** green on its own — affected-package typecheck clean and tests pass with 0 failures (verified across the full stack build).

## Summary by CodeRabbit

* **New Features**
  * Added a percentile calculation helper (`percentileOf`) with R-7 behavior and strict input validation.

* **Documentation**
  * Added a short documentation index in the README.
  * Added contribution guidance for the Bun workspace monorepo.
  * Added methodology documentation for dataset generation and reproducibility.
  * Added/expanded ADRs covering boundary parsing, enforced dependency DAG, PTS catalog drift gating, aggregation/promotion pipeline, and host vs effective spec handling.

* **Tests**
  * Extended tests for percentile edge cases and metric catalog invariants (ID format and derived-metric provenance).

### CODE REVIEW

**Comment by dbworku** [comment]
* **#77** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/77?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/77?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-ready</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/starslingdev?ref=merge-queue-instructions-in-stack-comment&prId=77).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/schema/src/analysis.ts [hunk 1]
```diff
@@ -39,6 +39,32 @@ function percentile(sorted: readonly number[], p: number): number {
...
 }
 
+/**
+ * Percentile `p` (0–1) of an unsorted, non-empty Sample set — the public entry point to the same R-7
+ * interpolation {@link aggregate} computes its p50/p95 with. Sorts a copy and defers to the private
+ * {@link percentile} order-statistic helper, so a one-off percentile and the Aggregates distribution
+ * share one code path and can never diverge. Throws on an empty set, a non-finite Sample, or a `p`
+ * outside [0, 1] (same fail-fast posture as {@link aggregate}); `p` of 0 and 1 resolve to min and max.
+ */
+export function percentileOf(samples: number[], p: number): number {
```
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/schema/src/analysis.ts]: Addressed / acknowledged in the latest revision.

File: docs/methodology.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+`detachedPoll`), and the harness selects a transport per step: a step that could outlast the provider's
+synchronous cap runs **detached + poll** where supported, everything else runs directly. This is why a
+multi-minute suite completes on a single-round-trip-capped provider (e.g. Daytona's server-side HTTP 408)
+without being Daytona-specific.
+
+## The dataset pipeline
+
+1. **Run** — `bench-suite <provider> <suite>` boots a sandbox, runs the suite's mise tasks, pulls the
+   raw tree (`data/raw/<runId>/<provider>/<suite>/`), and normalizes it into a Run document.
+2. **Matrix** — the `bench-matrix` workflow fans `plan-matrix` (provider × suite) out into one job per
```
[the remark this answers is not in the document]
Comment (dbworku) [reply docs/methodology.md:80]: Addressed / acknowledged in the latest revision.

File: docs/adr/0005-host-vs-effective-spec-split.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+silently mix "the sandbox you were sold" with "the box it happened to land on" — and an aggregate over
+shards scheduled on different hosts would average across hardware without anyone noticing.
+
+## Decision
+
+**Record observed specs as two explicit sides — effective and host — in one
+[`ObservedSpecs`](../../packages/schema/src/run.ts).** `vcpus`/`memoryGb`/`diskGb` are the EFFECTIVE
+sandbox size (the cgroup quota where enforced); `hostVcpus`/`hostMemoryGb` disclose the underlying
+machine when probes see through the boundary. `cpuModel`/`cpuMicroarch`/`cpuMhz` and friends are
+HOST-side by construction — `cpuMicroarch` is a generation label derived from the host `cpuModel`,
```
[the remark this answers is not in the document]
Comment (dbworku) [reply docs/adr/0005-host-vs-effective-spec-split.md:23]: Addressed / acknowledged in the latest revision.

File: docs/methodology.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbworku) [reply docs/methodology.md:80]: (correction) These entry points land in sibling stack PRs, not this docs PR: the `bench-matrix` workflow + `plan-matrix` bin come from ENG-66 (#69), the `leaderboard` bin from ENG-68 (#74, apps/cli bin "leaderboard"), and the `stability` bin from ENG-71 (#75, apps/cli bin "stability"). The methodology doc intentionally describes the full pipeline the stack delivers; no change needed here.

File: docs/adr/0005-host-vs-effective-spec-split.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbworku) [reply docs/adr/0005-host-vs-effective-spec-split.md:23]: (correction) Both fields are introduced by ENG-70 (#72, host-fingerprint): its packages/schema/src/run.ts adds `cpuMicroarch?: string` (host-side label derived from cpuModel) and `hostHeterogeneous?: boolean` (aggregate-only flag) to observedSpecsSchema, plus resolveCpuMicroarch in packages/results. The ADR documents the design the stack lands; it is accurate once #72 merges alongside this docs PR. No change needed.

File: packages/schema/src/analysis.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/schema/src/analysis.ts]: (correction) ✅ Addressed in the latest revision. Change the signature `export function percentileOf(samples: number[], p: number)` to `export function percentileOf(samples: readonly number[], p: number)`. Zero-risk: body already treats samples as read-only ([...samples].sort). Optionally widen aggregate(samples: number[]) too for API consistency, but percentileOf alone is a clean quick-win.

**Review by cosmic-emerald — APPROVED** [review]
